service/storage_proxy.cc

```diff
@@ -3296,6 +3308,25 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit), std::monostate(), is_cancellable::no);
 }
 
+// After sending the write to replicas(targets), the replicas might generate send view updates. If the number of view updates
+// that the replica is already sending is too high, the replica should not send any more view updates, and consequently,
+// we should not send the write to the replica.
+template<typename Range>
+bool storage_proxy::can_update_views(const Range& targets, const schema_ptr& s, db::write_type type, tracing::trace_state_ptr tr_state) const {
```

Comment:
The template here is unfortunate. It causes bloat and reduces performance.

I think the main problem is the use of boost::range::join(). It's also pretty bad as it generates complex iterators that then bloat everything that touches them. It's better to copy into a small_vector and then use std::span<> to communicate with other functions.

However, it's all a preexisting problem. We can ignore it for now but please do a cleanup pass later that removes the boost::range::join and the template.